jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js

```diff
@@ -0,0 +1,140 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('entity-resources', function() {
+
+  juju.components.EntityResources = React.createClass({
+
+    propTypes: {
+      apiVersion: React.PropTypes.string.isRequired,
+      charmId: React.PropTypes.string.isRequired,
+      charmstoreURL: React.PropTypes.string.isRequired,
+      getResources: React.PropTypes.func.isRequired,
+      pluralize: React.PropTypes.func.isRequired
+    },
+
+    /**
+      Get the current state.
+
+      @method getInitialState
+      @returns {String} The current state.
+    */
+    getInitialState: function() {
+      this.resourcesXHR = null;
+      return {
+        loading: false,
+        resources: null
+      };
+    },
+
+    componentDidMount: function() {
+      this._getResources(this.props.charmId);
+    },
+
+    componentWillUnmount: function() {
+      this.resourcesXHR.abort();
+    },
+
+    /**
+      Get a list of resources for the charm.
+
+      @method _getResources
+      @param {String} charmId The charm id.
+    */
+    _getResources: function(charmId) {
```

Comment:
Why did you have this as an additional call instead of adding 'resources' to the `getEntity` call in charmstore.js? Then the data would be just be available when rendering the entitydetails